src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_gligen.py

```diff
@@ -0,0 +1,826 @@
+            boxes = boxes.unsqueeze(0).expand(repeat_batch, -1, -1).clone()
+            text_embeddings = text_embeddings.unsqueeze(0).expand(repeat_batch, -1, -1).clone()
+            masks = masks.unsqueeze(0).expand(repeat_batch, -1).clone()
+            if do_classifier_free_guidance:
+                masks[: repeat_batch // 2] = 0
+            if cross_attention_kwargs is None:
+                cross_attention_kwargs = {}
+            cross_attention_kwargs["gligen"] = {"boxes": boxes, "positive_embeddings": text_embeddings, "masks": masks}
+
+            # Prepare latent variables for GLIGEN inpainting
+            if gligen_inpaint_image is not None:
+                if gligen_inpaint_image.size != (self.vae.sample_size, self.vae.sample_size):
+
+                    def crop(im, new_width, new_height):
+                        width, height = im.size
+                        left = (width - new_width) / 2
+                        top = (height - new_height) / 2
+                        right = (width + new_width) / 2
+                        bottom = (height + new_height) / 2
+                        return im.crop((left, top, right, bottom))
+
+                    def target_size_center_crop(im, new_hw):
+                        width, height = im.size
+                        if width != height:
+                            im = crop(im, min(height, width), min(height, width))
+                        return im.resize((new_hw, new_hw), PIL.Image.LANCZOS)
+
+                    gligen_inpaint_image = target_size_center_crop(gligen_inpaint_image, self.vae.sample_size)
+
+                gligen_inpaint_image = torch.from_numpy(np.asarray(gligen_inpaint_image))
+                gligen_inpaint_image = gligen_inpaint_image.unsqueeze(0).permute(0, 3, 1, 2)
+                gligen_inpaint_image = gligen_inpaint_image.to(dtype=torch.float32) / 127.5 - 1.0
+                gligen_inpaint_image = gligen_inpaint_image.to(dtype=self.vae.dtype, device=self.vae.device)
+                gligen_inpaint_latent = self.vae.encode(gligen_inpaint_image).latent_dist.sample()
+                gligen_inpaint_latent = self.vae.config.scaling_factor * gligen_inpaint_latent
+                gligen_inpaint_mask = draw_inpaint_mask_from_boxes(_boxes[:n_objs], gligen_inpaint_latent.shape[2:])
+                gligen_inpaint_mask = gligen_inpaint_mask.to(
+                    dtype=gligen_inpaint_latent.dtype, device=gligen_inpaint_latent.device
+                )
+                gligen_inpaint_mask = gligen_inpaint_mask[None, None]
+                gligen_inpaint_mask_addition = torch.cat(
+                    (gligen_inpaint_latent * gligen_inpaint_mask, gligen_inpaint_mask), dim=1
+                )
+                gligen_inpaint_mask_addition = gligen_inpaint_mask_addition.expand(repeat_batch, -1, -1, -1).clone()
+
+        num_grounding_steps = int(gligen_scheduled_sampling_beta * len(timesteps))
+        self.enable_fuser(True)
```

Comment:
Not a fan of calling functions inline like this? Could you implement this directly here?

Reply:
@sayakpaul I totally agree with you, but`self.enable_fuser()` is called again down below in this file.
If I implement it directly, I will have to add two copies of the same code, which might distract the readability.